sql3/test/defs/defs_date_functions.go

```diff
@@ -241,5 +241,49 @@ var datePartTests = TableTest{
 			),
 			Compare: CompareExactUnordered,
 		},
+		// dateTimeName tests
+		// make sure returning a string as the year still works
+		// if this works then the other parts converted to strings of digits should also work
+		{
+			SQLs: sqls(
+				"select _id, datetimename('yy', ts) from dateparttests",
```

Comment:
I am distrustful of `yy` as a name here. Do we have a spec for what the range names/types should be? Is there a reason `y` is duplicated? In particular, if I were going to see `yy`, I'd expect the result to be `12`, with `yyyy` producing `2012`.

Reply:
This is working off the constants provided in inbuiltfunctionsdate.go - they're inconsistent and cleaning them up would probably be a good idea, but I think that's a separate ticket since it'll touch a whole bunch of different stuff that's using them.